zerver/actions/users.py

```diff
@@ -245,6 +252,65 @@ def change_user_is_active(user_profile: UserProfile, value: bool) -> None:
         Subscription.objects.filter(user_profile=user_profile).update(is_user_active=value)
 
 
+def send_events_for_user_deactivation(user_profile: UserProfile) -> None:
+    event_deactivate_user = dict(
+        type="realm_user",
+        op="update",
+        person=dict(user_id=user_profile.id, is_active=False),
+    )
+    realm = user_profile.realm
+
+    if not user_access_restricted_in_realm(user_profile):
+        send_event_on_commit(realm, event_deactivate_user, active_user_ids(realm.id))
+        return
+
+    non_guest_user_ids = active_non_guest_user_ids(realm.id)
+    users_involved_in_dms_dict = get_users_involved_in_dms_with_target_users([user_profile], realm)
+
+    deactivated_user_subs = Subscription.objects.filter(
+        user_profile=user_profile,
+        recipient__type__in=[Recipient.STREAM, Recipient.HUDDLE],
+        active=True,
+    ).values_list("recipient_id", flat=True)
+    subscribers_in_deactivated_user_subs = Subscription.objects.filter(
+        recipient_id__in=list(deactivated_user_subs),
+        recipient__type__in=[Recipient.STREAM, Recipient.HUDDLE],
+        is_user_active=True,
+        active=True,
+    ).values_list("recipient__type", "user_profile_id")
```

Comment:
Did not use existing functions here because we need separate subscriptions for streams and huddles.

Reply:
Added this comment
```
+    # This code path is parallel to
+    # get_subscribers_of_target_user_subscriptions, but can't reuse it
+    # because we need to process stream and huddle subscriptions
+    # separately.
```